### PR TITLE
Update to PHP 8

### PR DESCRIPTION
**Pull request by settermjd**
This is a small PR designed to update the project to support the 8.0 release of PHP. It primarily updates composer.json so that the required packages work with the newer release. It also refactors a test which was failing.

### CODE REVIEW

File: tests/Functional/BomClientTest.php
Comment (kimpepper) [inline tests/Functional/BomClientTest.php:1]: Can you split out the changes to `testGetForecast` into a separate PR?

**Review by kimpepper — CHANGES_REQUESTED** [review]

File: tests/Functional/BomClientTest.php
Comment (settermjd) [reply tests/Functional/BomClientTest.php:1]: Done.

**Comment by settermjd** [comment]
I'll fix the build errors this weekend.

**Comment by kimpepper** [comment]
This is outdated now as PHP 8.1+ is supported.

**Pull request closed without merging**